nemo_skills/dataset/mmau-pro/prepare.py

```diff
@@ -0,0 +1,198 @@
+""",
+        "instruction_following": """# Instruction following questions evaluated with AIF format
+METRICS_TYPE = "speechlm"
+EVAL_ARGS = "++eval_type=mmau-pro"
+GENERATION_ARGS = "++prompt_format=openai"
+
+# Single chunk (small dataset)
+NUM_CHUNKS = 1""",
+    }
+
+
+def download_mmau_data(download_dir, hf_token):
+    """Download and extract MMAU-Pro data.zip file."""
+    download_dir = Path(download_dir)
+    download_dir.mkdir(parents=True, exist_ok=True)
+
+    data_zip_path = download_dir / "data.zip"
+    extracted_data_dir = download_dir / "data"
+
+    if extracted_data_dir.exists() and any(extracted_data_dir.iterdir()):
+        print(f"Data already exists at {extracted_data_dir}")
+        return extracted_data_dir
+
+    if not data_zip_path.exists():
+        print("Downloading MMAU-Pro data.zip...")
+        cmd = [
+            "wget",
+            "--header",
+            f"Authorization: Bearer {hf_token}",
+            "https://huggingface.co/datasets/gamma-lab-umd/MMAU-Pro/resolve/main/data.zip",
+            "-O",
+            str(data_zip_path),
+        ]
+        subprocess.run(cmd, check=True)
+
+    print("Extracting data.zip...")
+    with zipfile.ZipFile(data_zip_path, "r") as zip_ref:
+        zip_ref.extractall(download_dir)
+
+    return extracted_data_dir
+
+
+def format_entry(entry, with_audio=False):
+    """Format entry for nemo-skills with OpenAI messages and audio support."""
+    choices = entry.get("choices", []) or []
+
+    formatted_entry = {
+        "expected_answer": entry["answer"],
+        **get_mcq_fields(entry["question"], choices),
+        **{k: v for k, v in entry.items() if k not in ["answer"]},
+    }
+
+    category = entry.get("category", "")
+
+    if category == "open":
+        content = entry["question"]
+    elif choices and len(choices) > 1:
+        options_text = "\n".join(f"{chr(65 + i)}. {choice}" for i, choice in enumerate(choices))
+        content = f"{entry['question']}\n\n{options_text}"
+    else:
+        content = entry["question"]
+
+    user_message = {"role": "user", "content": content}
+
+    if entry.get("audio_path"):
+        audio_path = entry["audio_path"]
+
+        if isinstance(audio_path, list) and audio_path:
+            user_message["audios"] = [{"path": path, "duration": 10.0} for path in audio_path]
+        elif isinstance(audio_path, str):
+            user_message["audio"] = {"path": audio_path, "duration": 10.0}
+
+    formatted_entry["messages"] = [user_message]
+    return formatted_entry
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Prepare MMAU-Pro dataset for nemo-skills")
+    parser.add_argument("--split", default="test", choices=["validation", "test"])
+    parser.add_argument("--with-audio", action="store_true", help="Download audio files (requires HF_TOKEN)")
+    parser.add_argument("--download-dir", help="Directory for audio files (required with --with-audio)")
```

Comment:
can we instead reuse a common data_dir parameter of prepare_data / eval pipelines?